pkg/ipcache/kvstore.go

```diff
@@ -336,7 +336,7 @@ restart:
 
 				if m, ok := globalMap.marshaledIPIDPairs[event.Key]; ok {
 					log.WithField("ip", ip).Warning("Received kvstore delete notification for alive ipcache entry")
-					err := globalMap.store.upsert(event.Key, m, true)
+					err := globalMap.store.upsert(context.TODO(), event.Key, m, true)
```

Comment:
Can be handled in another PR, but why isn't the watcher also in a controller that could receive context?

Reply:
Agreed. Definitely can be done. The initial focus is on controllers that are attached to endpoints and thus need to be stopped and removed.